Libraries/plist/Sources/Format/BaseXMLParser.cpp

```diff
@@ -9,21 +9,284 @@
 
 #include <plist/Format/BaseXMLParser.h>
 
+#include <algorithm>
+#include <mutex>
 #include <cerrno>
 #include <cstdlib>
+#include <cstdarg>
+
+#if _WIN32
+#include <windows.h>
+#endif
 
 using plist::Format::BaseXMLParser;
 
+#if _WIN32
+using WideString = std::basic_string<std::remove_const<std::remove_pointer<LPCWSTR>::type>::type>;
```

Comment:
`std::wstring` exists on Unix too with different semantics; I'd prefer to avoid it completely than accidentally use it outside Windows.

I like `std::remove_pointer` and `std::remove_const` since they make it clear that the goal is a string of that type rather than assuming that `WCHAR` is the base, although I know it always is... yeah I guess there's no real point to it.